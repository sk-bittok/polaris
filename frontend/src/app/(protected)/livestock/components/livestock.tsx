import { Button } from "@/components/ui/button";
import {
  Dialog,
  DialogContent,
  DialogDescription,
  DialogHeader,
  DialogTitle,
  DialogTrigger,
} from "@/components/ui/dialog";
import { CircleAlert, Trash2 } from "lucide-react";
import React from "react";

export const InfoCard = ({ title, children, className = "" }) => (
  <div
    className={`bg-white dark:bg-gray-800 p-6 rounded-xl shadow-sm border-b border-gray-100 dark:border-gray-700 ${className}`}
  >
    <h3 className="text-lg font-semibold mb-4 text-gray-800 dark:text-gray-100">
      {title}
    </h3>
    {children}
  </div>
);

export const InfoRow = ({ label, value, className = "" }) => (
  <div className="flex justify-between items-center pb-2 gap-4 border-b border-gray-100 dark:border-gray-700">
    <span className="text-gray-500 dark:text-gray-400">{label}</span>
    <span
      className={`font-medium text-gray-800 dark:text-gray-100 ${className}`}
    >
      {value}
    </span>
  </div>
);

// Reusable EmptySection Component
export const EmptySection = ({ icon, message, description, buttonText }) => (
  <div className="bg-gray-50 dark:bg-gray-700 rounded-lg p-8 text-center">
    {icon && (
      <span className="mx-auto mb-2 text-gray-400 dark:text-gray-500 flex justify-center">
        {icon}
      </span>
    )}
    <p className="text-gray-500 dark:text-gray-400 text-sm">{message}</p>
    <p className="text-gray-400 dark:text-gray-500 text-xs mt-2 mb-4">
      {description}
    </p>
    {buttonText && (
      <Button variant="outline" size="sm" className="mt-3 text-xs">
        {buttonText}
      </Button>
    )}
  </div>
);

// Reusable Section with Header and Action Button
export const SectionWithAction = ({
  title,
  actionText,
  onAction,
  children,
}: {
  title: string;
  actionText: string;
  onAction: () => void;
  children: React.ReactNode;
}) => (
  <div className="bg-white dark:bg-gray-800 rounded-xl shadow-sm p-6 border border-gray-100 dark:border-gray-700">
    <div className="flex items-center justify-between mb-4">
      <h3 className="text-lg font-semibold text-gray-800 dark:text-gray-100">
        {title}
      </h3>
      <Button
        variant="outline"
        size="sm"
        className="text-blue-600 dark:text-blue-300"
        onClick={onAction}
      >
        {actionText}
      </Button>
    </div>
    {children}
  </div>
);

export const DeleteDialog = ({
  confirmDelete,
  children,
}: {
  confirmDelete: () => void;
  children: React.ReactNode;
}) => (
  <Dialog>
    <DialogTrigger asChild>{children}</DialogTrigger>
    <DialogContent className="sm:max-w-[435px]">
      <DialogHeader>
        <DialogTitle>Remove animal</DialogTitle>
        <DialogDescription>
          <span className="flex items-center mb-4">
            <CircleAlert size={24} className="mr-2" />
            This action cannot be undone. Are you sure you want to permanently
            delete this record from our servers?
          </span>
        </DialogDescription>
      </DialogHeader>

      <div className="flex items-center justify-end">
        <Button
          onClick={confirmDelete}
          type="button"
          className="flex items-center px-4 py-2 bg-red-600 text-white rounded-md hover:bg-red-700 transition-colors"
        >
          <Trash2 className="mr-2" size={20} />
          Delete
        </Button>
      </div>
    </DialogContent>
  </Dialog>
);
