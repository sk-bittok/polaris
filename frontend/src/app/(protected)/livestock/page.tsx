'use client';

import { useGetLivestockQuery } from "@/state/api"
import Link from "next/link";
import { formatDistanceToNow } from "date-fns";
import { Plus } from "lucide-react";
import { Button } from "@/components/ui/button";

export default function LivestockListPage() {
  const { data, isError, isLoading, isSuccess, error } = useGetLivestockQuery();

  if (isError && data === undefined) {
    return (
      <div className="flex items-center justify-center h-screen">
        <div className="bg-red-50 dark:bg-red-900 p-8 rounded-lg shadow-md text-center">
          {error.error ? (<>
            <h2 className="text-red-700 dark:text-red-300 text-2xl font-bold">Error: 500</h2>
            <p className="text-red-500 dark:text-red-400">An error occurred: {error.error}</p>
          </>
          ) : (
            <>
              <h2 className="text-red-700 dark:text-red-300 text-2xl font-bold">Error: {error?.status}</h2>
              <p className="text-red-500 dark:text-red-400">An error occurred: {error?.data.message}</p>
            </>
          )}
          <Button className="mt-4 bg-red-600 hover:bg-red-700 text-white">
            <Link href="/livestock">Return to Livestock</Link>
          </Button>
        </div>
      </div>
    );
  }


  return (
    <div className="container mx-auto px-4 py-8">
      <div className="flex justify-between items-center mb-6">
        <h1 className="text-2xl font-bold">Livestock</h1>
        <Link href="/livestock/new" className="flex items-center bg-blue-500 hover:bg-blue-600 px-4 py-2 rounded-md">
          <Plus className="mr-2 w-5 h-5 font-bold" />
          Add new
        </Link>
      </div>

      {isLoading ? (
        <div className="flex justify-center items-center min-h-screen">
          <div className="animate-spin rounded-full h-12 w-12 border-t-2 border-b-2 border-blue-500"></div>
        </div>
      ) : isSuccess && data.length === 0 ? (
        <div className="text-center py-12 bg-gray-50 rounded-lg">
          <p className="text-gray-500">No livestock yet.</p>
        </div>
      ) : (
        <div className="overflow-y-auto">
          <table className="min-w-full bg-white border border-gray-200">
            <thead>
              <tr className="bg-gray-100">
                <th className="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider">ID</th>
                <th className="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider">Tag ID</th>
                <th className="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider">Name</th>
                <th className="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider">Category</th>
                <th className="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider">Breed</th>
                <th className="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider">Gender</th>
                <th className="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider">Status</th>
                <th className="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider">Updated</th>
                <th className="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider">Status</th>
              </tr>
            </thead>
            <tbody className="divide-y divide-gray-200">
              {data.map((livestock) => (
                <tr key={livestock.pid} className="hover:bg-gray-50">
                  <td className="px-6 py-4 whitespace-nowrap text-sm text-gray-500">{livestock.id}</td>
                  <td className="px-6 py-4 whitespace-nowrap text-sm text-gray-500">{livestock.tagId}</td>
                  <td className="px-6 py-4 whitespace-nowrap text-sm font-medium text-gray-900">{livestock.name}</td>
                  <td className="px-6 py-4 whitespace-nowrap text-sm text-gray-500">{livestock.specieName}</td>
                  <td className="px-6 py-4 whitespace-nowrap text-sm text-gray-500">{livestock.breedName}</td>
                  <td className="px-6 py-4 whitespace-nowrap text-sm text-gray-500">{livestock.gender}</td>
                  <td className="px-6 py-4 whitespace-nowrap">
                    <span className={`px-2 inline-flex text-xs leading-5 font-semibold rounded-full ${livestock.status === 'Active' ? 'bg-green-100 text-green-800' :
                      livestock.status === 'Sold' ? 'bg-blue-100 text-blue-800' :
                        livestock.status === 'Deceased' ? 'bg-red-100 text-red-800' :
                          'bg-gray-100 text-gray-800'
                      }`}>
                      {livestock.status}
                    </span>
                  </td>
                  <td className="px-6 py-4 whitespace-nowrap text-sm text-gray-500">
                    {formatDistanceToNow(new Date(livestock.updatedAt), { addSuffix: true })}
                  </td>
                  <td className="px-6 py-4 whitespace-nowrap text-sm text-gray-500">
                    <Link
                      href={`/livestock/${livestock.id}`}
                      className="text-blue-600 hover:text-blue-900 mr-3"
                    >
                      View
                    </Link>
                    <Link
                      href={`/animals/${livestock.id}/edit`}
                      className="text-indigo-600 hover:text-indigo-900"
                    >
                      Edit
                    </Link>
                  </td>
                </tr>
              ))}
            </tbody>
          </table>
        </div>
      )}
    </div>
  );
}
