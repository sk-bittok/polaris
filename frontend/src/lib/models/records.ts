export interface ProductionRecord {
	id: number;
	livestockName: string;
	animalPid: string;
	productType: string;
	unit: string;
	quantity: string;
	notes?: string;
	quality?: string;
	recordDate?: Date;
	createdAt: Date;
	createdBy: string;
	createdByName: string;
	updatedAt: Date;
}

export interface NewProductionRecord {
	id: number;
	animal_pid: string;
	organisation_pid: string;
	product_type: string;
	quantity: number;
	unit: string;
	record_date?: string;
	quality?: string;
	notes?: string;
	created_by: string;
	created_at: Date;
	updated_at: Date;
}

export interface HealthRecordResponse {
	id: number;
	organisationPid: string;
	organisationName: string;
	animalName: string;
	animalPid: string;
	animalTagId: string;
	recordType: string;
	recordDate: Date;
	description: string;
	treatment: string;
	medicine?: string;
	dosage?: string;
	cost?: number;
	notes?: string;
	performedBy?: string;
	createdBy: string;
	createdByName: string;
	createdAt: Date;
	updatedAt: Date;
}

export interface HealthRecord {
	id: number;
	organisation_pid: string;
	animal_pid: string;
	record_type: string;
	record_date: Date;
	description: string;
	treatment: string;
	medicine?: string;
	dosage?: string;
	cost?: number;
	notes?: string;
	performed_by?: string;
	created_by: string;
	created_at: Date;
	updated_at: Date;
}

export interface WeightRecord {
	id: number;
	organisation_pid: string;
	animal_pid: string;
	mass: number;
	record_date: string;
	notes?: string;
	created_by: string;
	created_at: Date;
	updated_at: Date;
}

export interface WeightRecordResponse {
	id: number;
	organisationPid: string;
	organisationName: string;
	animalPid: string;
	animalName: string;
	animalTagId: string;
	mass: number;
	recordDate: string;
	notes?: string;
	createdBy: string;
	createdByName: string;
	created_at: Date;
	updated_at: Date;
}
