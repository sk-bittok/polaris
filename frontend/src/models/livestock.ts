
export interface Breed {
  id: number;
  organisationPid?: string;
  specie: Category;
  name: string;
  maleWeightRange?: string;
  femaleWeightRange?: string;
  gestationPeriod?: string;
  description?: string;
  createdAt: string;
  isSystemDefined: boolean;
}

export enum Category {
  Cattle = "cattle",
  Sheep = "sheep",
  Goat = "goats",
  Chicken = "chicken",
  Pig = "pigs"
}

export enum Gender {
  Male = "male",
  Female = "female",
  Unkown = "unkown",
}

export enum Status {
  Active = "active",
  Sold = "sold",
  Deceased = "deceased",
  Transferred = "transferred"
}

export interface RegisterLivestock {
  tagId: string;
  name: string;
  gender: string;
  status: string;
  breed: string;
  specie: string;
  dateOfBirth?: string | Date | null;
  weightAtBirth?: number | null;
  maleParentId?: string | null;
  femaleParentId?: string | null;
  purchasePrice?: number | null;
  purchaseDate?: string | Date | null;
  currentWeight?: number | null;
  notes?: string | null;
};

export interface UpdateLivestock {
  tagId?: string | null;
  name?: string | null;
  gender?: string | null;
  status?: string | null;
  breed?: string | null;
  specie?: string | null;
  dateOfBirth?: string | Date | null;
  weightAtBirth?: number | null;
  maleParentId?: string | null;
  femaleParentId?: string | null;
  purchasePrice?: number | null;
  purchaseDate?: string | Date | null;
  currentWeight?: number | null;
  notes?: string | null;
};

export interface Livestock {
  id: number;
  pid: string;
  organisationName: string;
  tagId: string;
  name: string;
  breedName: string;
  specieName: Category;
  dateOfBirth?: Date;
  gender: Gender;
  parentMaleName?: string;
  parentFemaleName?: string;
  status: Status;
  parentMaleTagId?: string;
  parentFemaleTagId?: string;
  purchaseDate?:  Date;
  purchasePrice?: string;
  weightAtBirth?: string;
  currentWeight?: string;
  notes?: string;
  createdByName: string;
  createdAt: Date;
  updatedAt: Date;
}

