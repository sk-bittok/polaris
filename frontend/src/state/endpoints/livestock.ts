import type { LinkOffspring } from "@/lib/schemas/animal";
import type {
	Livestock,
	RegisterLivestock,
	UpdateLivestock,
} from "@/models/livestock";
import type { EndpointBuilder } from "@reduxjs/toolkit/query";

export const livestockEndpoints = (build: EndpointBuilder<any, any, any>) => ({
	getLivestock: build.query<Livestock[], void>({
		query: () => ({ url: "/animals" }),
		providesTags: ["GetLivestock"],
	}),
	getLivestockDescendants: build.query<Livestock[], string>({
		query: (params) => ({ url: `/animals?${params}` }),
		providesTags: ["GetLineageRecords"],
	}),
	registerLivestock: build.mutation<Livestock, RegisterLivestock>({
		query: (json) => ({
			url: "/animals",
			method: "POST",
			body: json,
		}),
		invalidatesTags: ["GetLivestock"],
	}),
	getLivestockById: build.query<Livestock, string>({
		query: (id) => ({ url: `/animals/${id}` }),
		providesTags: (result, error, id) => [{ type: "GetLivestock", id }],
	}),
	getLivestockByTagId: build.query<Livestock, string>({
		query: (tagId) => ({ url: `animals/tag-id/${tagId}` }),
		providesTags: (result, error, id) => [{ type: "GetLivestock", id }],
	}),
	deleteLivestockById: build.mutation<void, string>({
		query: (id) => ({
			url: `/animals/${id}`,
			method: "DELETE",
		}),
		invalidatesTags: ["GetLivestock"],
	}),
	updateLivestockById: build.mutation<
		Livestock,
		{ data: UpdateLivestock; id: string }
	>({
		query: ({ data, id }) => ({
			url: `/animals/${id}`,
			method: "PATCH",
			body: data,
		}),
		invalidatesTags: ["GetBreeds"],
	}),
	linkOffspring: build.mutation<Livestock, LinkOffspring>({
		query: (data) => ({
			url: "/animals/link-offspring",
			method: "PATCH",
			body: data,
		}),
		invalidatesTags: ["GetLineageRecords"],
	}),
});
