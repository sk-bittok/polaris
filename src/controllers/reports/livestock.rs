use axum::{
    Json, Router, debug_handler,
    extract::State,
    http::StatusCode,
    response::{IntoResponse, Response},
    routing::{get, post},
};

use crate::{
    AppContext, Result,
    models::{livestock::LivestockSummary, users::User},
};

#[debug_handler]
async fn all(State(ctx): State<AppContext>, user: User) -> Result<Response> {
    let reports = LivestockSummary::find_all(&ctx.db, user.organisation_pid).await?;

    Ok((StatusCode::OK, Json(reports)).into_response())
}

#[debug_handler]
async fn add(State(ctx): State<AppContext>, user: User) -> Result<Response> {
    let report = LivestockSummary::generate(&ctx.db, user.organisation_pid).await?;

    Ok((StatusCode::CREATED, Json(report)).into_response())
}

pub fn router(ctx: AppContext) -> Router {
    Router::new()
        .route("/", get(all))
        .route("/", post(add))
        .with_state(ctx)
}
