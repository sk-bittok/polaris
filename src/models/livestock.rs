#![allow(clippy::missing_errors_doc)]
#![allow(clippy::needless_raw_string_hashes)]

use chrono::{DateTime, FixedOffset};
use rust_decimal::Decimal;
use serde::{Deserialize, Serialize};
use sqlx::{Encode, Executor, Postgres, prelude::FromRow};
use uuid::Uuid;

use super::ModelResult;

#[derive(Debug, Deserialize, FromRow, Encode, Serialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct LivestockSummary {
    pub pid: Uuid,
    pub organisation_pid: Uuid,
    pub id: i32,
    pub total: i32,
    pub males: i32,
    pub females: i32,
    pub unkown_gender: i32,
    pub active: i32,
    pub transferred: i32,
    pub sold: i32,
    pub deceased: i32,
    pub species: i32,
    pub breeds: i32,
    pub total_purchased_value: Decimal,
    pub created_at: DateTime<FixedOffset>,
    pub updated_at: DateTime<FixedOffset>,
}

#[derive(Debug, Deserialize, Serialize, FromRow, Encode, Clone)]
pub struct SummaryData {
    pub total: i64,
    pub males: i64,
    pub females: i64,
    pub unkown_gender: i64,
    pub active: i64,
    pub transferred: i64,
    pub sold: i64,
    pub deceased: i64,
    pub species: i64,
    pub breeds: i64,
    pub total_purchased_value: Decimal,
}

impl SummaryData {
    pub async fn find_by_organisation<'e, C>(db: &C, org_pid: Uuid) -> ModelResult<Option<Self>>
    where
        for<'a> &'a C: Executor<'e, Database = Postgres>,
    {
        let summary_data = sqlx::query_as::<_, Self>(
            r#"
            SELECT
                COUNT(*) as total,
                COUNT(*) FILTER (WHERE gender = 'male') as males,
                COUNT(*) FILTER (WHERE gender = 'female') as females,
                COUNT(*) FILTER (WHERE gender = 'unkown') as unkown_gender,
                COUNT(*) FILTER (WHERE status = 'active') as active,
                COUNT(*) FILTER (WHERE status = 'transferred') as transferred,
                COUNT(*) FILTER (WHERE status = 'deceased') as deceased,
                COUNT(*) FILTER (WHERE status = 'sold') as sold,
                COUNT(DISTINCT specie_id) as species,
                COUNT(DISTINCT breed_id) as breeds,
                COALESCE(SUM(purchase_price), 0) as total_purchased_value
            FROM animals
            WHERE organisation_pid = $1
            "#,
        )
        .bind(org_pid)
        .fetch_optional(db)
        .await?;

        Ok(summary_data)
    }
}

impl LivestockSummary {
    pub async fn generate<'e, C>(db: &C, org_pid: Uuid) -> ModelResult<Self>
    where
        for<'a> &'a C: Executor<'e, Database = Postgres>,
    {
        let summary_data = SummaryData::find_by_organisation(db, org_pid).await?;

        // Insert the data
        let query = sqlx::query_as::<_, Self>(
            r"
            INSERT INTO livestock_summary (
                organisation_pid,
                total,
                males,
                females,
                unkown_gender,
                active,
                transferred,
                sold,
                deceased,
                species,
                breeds,
                total_purchased_value
            )
            VALUES (
                $1,
                $2,
                $3,
                $4,
                $5,
                $6,
                $7,
                $8,
                $9,
                $10,
                $11,
                $12
            )
            RETURNING *
        ",
        )
        .bind(org_pid)
        .bind(summary_data.as_ref().map_or(0, |data| data.total))
        .bind(summary_data.as_ref().map_or(0, |data| data.males))
        .bind(summary_data.as_ref().map_or(0, |data| data.females))
        .bind(summary_data.as_ref().map_or(0, |data| data.unkown_gender))
        .bind(summary_data.as_ref().map_or(0, |data| data.active))
        .bind(summary_data.as_ref().map_or(0, |data| data.transferred))
        .bind(summary_data.as_ref().map_or(0, |data| data.sold))
        .bind(summary_data.as_ref().map_or(0, |data| data.deceased))
        .bind(summary_data.as_ref().map_or(0, |data| data.species))
        .bind(summary_data.as_ref().map_or(0, |data| data.breeds))
        .bind(
            summary_data
                .as_ref()
                .map_or_else(|| Decimal::new(000, 2), |data| data.total_purchased_value),
        )
        .fetch_one(db)
        .await?;

        Ok(query)
    }
}
