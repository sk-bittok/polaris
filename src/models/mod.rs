pub mod animals;
pub mod breeds;
pub mod dto;
pub mod enums;
pub mod errors;
pub mod health;
pub mod livestock;
pub mod orgs;
pub mod production;
pub mod roles;
pub mod species;
pub mod summaries;
pub mod users;
pub mod weight;

pub use self::{
    errors::{ModelError, ModelResult},
    summaries::*,
};
