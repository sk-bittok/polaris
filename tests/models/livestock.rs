use insta::{Settings, assert_debug_snapshot, with_settings};
use polaris::models::livestock::{LivestockSummary, SummaryData};
use rstest::rstest;
use serial_test::serial;
use uuid::Uuid;

use crate::{boot_test, cleanup_date, cleanup_int, cleanup_uuid, seed_data};

macro_rules! configure_insta {
    ($(expr:expr),*) => {
        let mut settings = Settings::clone_current();
        settings.set_prepend_module_to_snapshot(false);
        settings.set_snapshot_suffix("livestock");
        settings.set_snapshot_path("snapshots/summaries");
        let _guard = settings.bind_to_scope();
    };
}

#[rstest]
#[case(
    "can_find_summary_org_a",
    Uuid::parse_str("9d5b0c1e-6a48-4bce-b818-dc8c015fd8a0").unwrap(),
)]
#[case(
    "can_find_summary_org_b",
    Uuid::parse_str("4a93f0a8-4a91-482d-92d8-f0b3b084c2e4").unwrap(),
)]
#[tokio::test]
#[serial]
async fn can_find_summary_data(#[case] test_name: &str, #[case] org_pid: Uuid) {
    configure_insta!();

    let ctx = boot_test().await.unwrap();
    seed_data(&ctx.db).await.unwrap();

    let result = SummaryData::find_by_organisation(&ctx.db, org_pid).await;

    assert_debug_snapshot!(test_name, result);
}

#[rstest]
#[case(
    "can_generate_summary_org_a",
    Uuid::parse_str("9d5b0c1e-6a48-4bce-b818-dc8c015fd8a0").unwrap(),
)]
#[case(
    "can_generate_summary_org_b",
    Uuid::parse_str("4a93f0a8-4a91-482d-92d8-f0b3b084c2e4").unwrap(),
)]
#[tokio::test]
#[serial]
async fn can_generate_summary(#[case] test_name: &str, #[case] org_pid: Uuid) {
    configure_insta!();

    let ctx = boot_test().await.unwrap();
    seed_data(&ctx.db).await.unwrap();

    let result = LivestockSummary::generate(&ctx.db, org_pid).await;

    with_settings!({
        filters => {
            let mut filters = cleanup_date().to_vec();
            filters.extend(cleanup_uuid().to_vec());
            filters.extend(cleanup_int().to_vec());
            filters
        }
    }, {
        assert_debug_snapshot!(test_name, result);
    });
}
