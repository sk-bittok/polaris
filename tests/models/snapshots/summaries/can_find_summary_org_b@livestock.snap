---
source: tests/models/livestock.rs
assertion_line: 38
expression: result
---
Ok(
    Some(
        SummaryData {
            total: 9,
            males: 3,
            females: 6,
            unkown_gender: 0,
            active: 6,
            transferred: 1,
            sold: 1,
            deceased: 1,
            species: 1,
            breeds: 1,
            total_purchased_value: 55800.00,
        },
    ),
)
